packages/vite/src/node/plugins/resolve.ts

```diff
@@ -899,8 +906,23 @@ export async function tryOptimizedResolve(
     // lazily initialize resolvedSrc
     if (resolvedSrc == null) {
       try {
-        // this may throw errors if unable to resolve, e.g. aliased id
-        resolvedSrc = normalizePath(resolveFrom(id, path.dirname(importer)))
+        const { target: ssrTarget } = resolveOptions.ssrConfig ?? {}
+        const targetWeb = !ssr || ssrTarget === 'webworker'
+        const resolved = tryNodeResolve(
```

Comment:
I am not very sure this `tryOptimizedResolve` call would trigger some edge cases that return unexpected result 🤔.

I don't want `tryOptimizedResolve` to run after this line `const processResult = ......`. Should I add a new argument to `tryOptimizedResolve` to make it return early?

Reply:
> I am not very sure this `tryNodeResolve ` call would trigger some edge cases that return unexpected result 🤔.
> 
> I don't want `tryNodeResolve ` to run code after this line:
> 
> https://github.com/vitejs/vite/blob/fff689fa2c7641572f8ff6f2bdf48d2023034e03/packages/vite/src/node/plugins/resolve.ts#L747
> 
> . Should I add a new argument to `tryNodeResolve ` to make it return early? My worry is that `tryNodeResolve` already has too many arguments...

Will be resolved by this comment: https://github.com/vitejs/vite/pull/11410#issuecomment-1356823806